web-common/src/runtime-client/watch-request-client.ts

```diff
@@ -0,0 +1,82 @@
+import type {
+  V1WatchFilesResponse,
+  V1WatchLogsResponse,
+  V1WatchResourcesResponse,
+} from "@rilldata/web-common/runtime-client/index";
+import { streamingFetchWrapper } from "@rilldata/web-common/runtime-client/fetch-streaming-wrapper";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import type { Runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import { asyncWait } from "@rilldata/web-local/lib/util/waitUtils";
+import { get, Unsubscriber } from "svelte/store";
+
+type WatchResponse =
+  | V1WatchFilesResponse
+  | V1WatchResourcesResponse
+  | V1WatchLogsResponse;
+
+type StreamingFetchResponse<Res extends WatchResponse> = {
+  result?: Res;
+  error?: { code: number; message: string };
+};
+
+export class WatchRequestClient<Res extends WatchResponse> {
+  private controller: AbortController;
+
+  public constructor(
+    private readonly getUrl: (runtime: Runtime) => string,
```

Comment:
Instead of passing in a `getURL` function, it looks like you could just pass a `slug` (`/files/watch` or `/resources/watch`) and use that to compute the URL within the `WatchRequestClient`.

Reply:
Let this be as is. The way the url is built should be the responsibility of the caller. This way a single class can handle requests to runtime and admin name spaces.

Reply:
But this is part of the `runtime-client` and won't be used to connect to the admin server, right?

Reply:
Hmmm you are right, even the logs API is on the runtime. We can refactor to have just the final url path once we know of all end points.